### DISCUSSION TITLE
SH1106 mit Fusion Board

### DISCUSSION THREAD

**Discussion by spiderhl53** [hoylabs/OpenDTU-OnBattery#1337] [Q&A]
Ich habe folgende Konfiguration:
Fusion Board V2
OpenDTU-OnBattery aktuelle Version
2x Victron MTTP
Jk-BMS über RS485
Display SH1106
Hoymiles 1500
Ich habe in die 3V3-Leitung des Displays einen Schalter eingebaut, da ich das Display nur selten benötige. Wenn der Schalter geschlossen ist, ist alles ok. Wenn ich ihn jedoch öffne, ist nicht nur das Display aus (wie gewünscht), sondern es werden auch keine Daten mehr vom Inverter empfangen. Was läuft schief?

Gruß spiderHL

**Comment by stefan123t** [comment]
Vermutlich will die OpenDTU Daten an das Display senden und bekommt keine Verbindung / Bestätigung. Dass Du das Display einfach physisch / elektronisch abschaltest ist eine Sache, aber die Programmlogik muss das auch erwarten oder überspringen können. Hier fehlt vermutlich eine Fehlerbehandlung.

Einfach mal die Web Serial Console offen lassen und das Display ausschalten ?

**Reply by stefan123t** [reply]
Hier ist die Anleitung zur Web Serial Console: https://www.opendtu.solar/firmware/configuration/console/
bzw. zum USB Serial Debugging https://www.opendtu.solar/firmware/howto/serial_console/

**Comment by spiderhl53** [comment]
Hallo Stefan,
ich denke, dass du Recht hast. Wenn ich das Display ausschalte, kommen keine weiteren Meldungen mehr auf der Konsole, schalte ich es wieder ein, wird alles fortgesetzt.
Wenn ich das Display nicht anschließe ohne die Konfiguration zu ändern, läuft ebenfalls alles normal.
D.h. nur bei angeschlossenem Display und unterbrochener Stromversorgung (Schalter) des Displays bleibt das Programm offensichtlich hängen.

**Reply by schlimmchen** [reply]
> kommen keine weiteren Meldungen mehr auf der Konsole

Keine? Wenn der ESP32 sich verklemmt, weil die I2C Transaktion nicht gemacht werden kann, dann sollte eigentlich der Watchdog den ESP32 neu starten. Tut er das nicht?

Es kann natürlich sein, dass einer der Logik Pins dann Strom in die falsche Richtung "zieht" und dass dann der Core an sich verklemmt.

Lästiges Problem, aber sehr speziell. Helfen wird hier schwierig bzw. schnell sehr aufwändig. Kannst du vielleicht den Stecker des Displays ziehen? Mit konfiguriertem aber gänzlich abgeklemmten Display sollte die OpenDTU-OnBattery doch starten?

**Reply by spiderhl53** [reply]
Wie schon oben beschrieben:
Display angesteckt, Schalter ein     normales Verhalten
Display angesteckt, Schalter aus     keine weiteren Meldungen auf der Konsole, keine Aktualisierung der Livedaten
Schalter wieder an                           es kommen wieder Meldungen, Livedaten werden wieder aktualisiert
Display abgesteckt                          normales Verhalten

Da ich das OpenDTU-Gerät in der Garage hängen habe, wäre es nett gewesen, dort eine Anzeige zu haben, Diese wird aber 99% der Zeit nicht benötigt.
Ich werde mir nun damit behelfen, dass ich zwischen zwei Profilen (mit und ohne Displaydefinition) umschalte, das funktioniert ohne Probleme.

Dabei ist mir aufgefallen, dass die Ausrichtung des Displays (Rotation) bei Neustart oder Umschalten der Profile sich manchmal ändert. Eine Systematik konnte ich hierbei noch nicht feststellen.
